Library/Homebrew/utils/github/api.rb

```diff
@@ -65,16 +65,23 @@ class AuthenticationFailedError < Error
       def initialize(github_message)
         @github_message = github_message
         message = +"GitHub API Error: #{github_message}\n"
-        message << if Homebrew::EnvConfig.github_api_token
+        message << case GitHub::API.credentials_type
+        when :github_cli_token
           <<~EOS
-            HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
-              #{Formatter.url("https://github.com/settings/tokens")}
+            Your GitHub CLI login session may be invalid.
+            Refresh it with:
+              gh auth login -h github.com
           EOS
-        else
+        when :keychain_username_password
           <<~EOS
             The GitHub credentials in the macOS keychain may be invalid.
             Clear them with:
               printf "protocol=https\\nhost=github.com\\n" | git credential-osxkeychain erase
+          EOS
+        else
+          <<~EOS
+            HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
+              #{Formatter.url("https://github.com/settings/tokens")}
             #{GitHub.pat_blurb}
```

Comment:
I moved the `pat_blurb` here as I don't think it really makes sense with `keychain_username_password` -- instructions on creating a PAT and assiging it to `HOMEBREW_GITHUB_API_TOKEN` make more sense in the context of that token.

If a user has an invalid set of creds in the keychain, they will first be prompted to erase them. Then, if they still need credentials (e.g. for a private repo), they'll get clear instructions on how to create the token. This way the keychain functionality remains largely in the background.

Reply:
Will this not now be output even if `Homebrew::EnvConfig.github_api_token` is unset? If so, that's undesirable.

Reply:
The prior behavior was to complain of invalid credentials in the keychain and suggest creating a PAT and storing it in `HOMEBREW_GITHUB_API_TOKEN`; the fact that the keychain was referred to at all is unexpected for most users I think. Is this not a bit more consistent?

Reply:
I think an extra case checking if `if Homebrew::EnvConfig.github_api_token.present?` and saying `HOMEBREW_GITHUB_API_TOKEN is unset` or similar would improve this, thanks!